lib/temporalio/worker.rb

```diff
@@ -41,38 +48,67 @@ def initialize(
       end
     end
 
-    def run
-      Async { |task| start(task) }
+    def run(&block)
+      Runner.new(self).run(&block)
     end
 
-    def start(reactor = nil)
+    def start(runner = nil)
```

Comment:
Should we discourage `start` from user use because they can't react to fatal worker errors like they can in `run`? Or am I misunderstanding?

Reply:
Yes, exactly. Basically I have to keep it public, because we're using it when starting multiple workers via the `Worker::Runner`, but the user is discouraged from using it.

Reply:
Can document that away and/or change the name to `start_and_kill_thread_on_error` to be unwieldy :-) Your call.

Reply:
Haha :) I've marked it here — https://github.com/temporalio/sdk-ruby/pull/100/files#diff-913ed58d77d1d4e414bfe371e4670b2f60cd30c4c94e0ed3dd39ad7e7fa7ad3dR92. Hope that's discouraging enough :)

Reply:
Is there a way to keep this method and shutdown private with passing around closures instead of the actual object?

Reply:
I'm not sure that would work with starting multiple workers together without a huge strain on the code readability (class/static methods are considered to be outside of the `private` scope in Ruby). I think a note in the docs is enough for now